tests/app/clients/test_twilio.py

```diff
@@ -129,6 +130,8 @@ def test_send_sms_call_with_sender_id_and_specifics(sample_service, notify_api,
             sms_sender_id=sms_sender_id
         )
 
+    assert response_dict['sid'] == reference
```

Comment:
This does not verify the response.

Reply:
when `twilio_sms_client.send_sms(...)` is called it will return the Twilio response `message.sid` which gets assigned to `reference`. I think it does, but it's confusing. I should probably change the variable name to make it more clear.

Reply:
I'm just not understanding how calling `send_sms` yields the sid "MMXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX". I don't see the mock. I'm missing something.

Reply:
The mocked return from Twilio is on line 114. 
`request_mock.post("https://api.twilio.com/2010-04-01/Accounts/TWILIO_TEST_ACCOUNT_SID_XXX/Messages.json", json=response_dict, status_code=200)` 

It includes the `response_dict`, which contains the sid. When `send_sms` is called it will then return the `message.sid` from the mocked Twilio response.

Reply:
I'm not sure I follow, either. 

Also `assert response_dict['sid'] == reference` is going to throw a fun error if that key doesn't exist. Maybe instead use `assert response_dict.get('sid') == reference` which at least returns None if for some reason the response doesn't have that key. The test will fail as expected but at least an exception won't be raised.